salestax.rb

```diff
@@ -0,0 +1,52 @@
+input = []
+ans = 'y'
+non_taxable = ["book", "chocolate","chocolates", "pills"]
+
+while ans != 'n' do
+  i = gets()
+  input.push(i)
+  puts "more items in basket?"
+  ans = gets()
+  ans = ans.chomp
+end 
+
+total = 0
+tax = 0
+flag = 0
+
+input.each do |i|
+  flag = 0
+  item_tax = 0
+  output = i.split()	
+  qty = (output[0]).chomp.to_i
+  amt = (output[-1]).chomp.to_f
+  item_total = qty*amt
+  if qty <= 0 || amt <=0.0
+    puts "invalid input"
+    break
+  end
+  item_tax += item_total*0.05 if output.include?("imported")
+  #puts " item tax1: #{item_tax}"
+  output.each do |o|
+    #o == "at" ? (print ":") : (print "#{o} ")
+    if non_taxable.include?(o)
+      flag = 1
+      break
+    end
+  end
+  output.pop()
+  output.pop
+  output.each { |o| print "#{o} "}
+  item_tax += item_total*0.1 if flag == 0
+  #puts " item tax2: #{item_tax}"
```

Comment:
What's this for?

Reply:
I'm checking if an item in the basket belongs to either of the non-taxable item list
so if it finds a keyword, say book, then it will set the flag to 1. So after the loop ends if the flag is 0 only then the tax will be added